FindCACTUS.cmake

```diff
@@ -1,145 +1,232 @@
+            cactus_${name}
+            NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH
+            HINTS ENV XDAQ_ROOT
+            PATHS /opt/cactus/
+            PATH_SUFFIXES lib lib64
+            DOC "Path of CACTUS library ${name}")
+
+        mark_as_advanced(CACTUS_${name}_LIBRARY)
+
+        # Try to find the headers
+        find_path(
+            CACTUS_INCLUDE_DIRS
+            ${cactus_${name}_header}
+            NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH
+            HINTS ENV XDAQ_ROOT
+            PATHS /opt/cactus/
+            PATH_SUFFIXES include
+            DOC "CACTUS include directory")
+
+        mark_as_advanced(CACTUS_INCLUDE_DIRS)
+
+        if(CACTUS_${name}_LIBRARY AND CACTUS_INCLUDE_DIRS)
+            # Found!
+            set(CACTUS_${name}_FOUND TRUE)
+
+            # Create the target
+            add_library(CACTUS::${name} SHARED IMPORTED)
+
+            # Set location
+            set_property(
+                TARGET CACTUS::${name}
+                PROPERTY IMPORTED_LOCATION
+                ${CACTUS_${name}_LIBRARY})
+
+            # Handle NO_SONAME
+            if(cactus_${name}_nosoname)
+                set_property(TARGET CACTUS::${name} PROPERTY IMPORTED_NO_SONAME TRUE)
+            endif()
+
+            # Set include path
+            set_target_properties(
+                CACTUS::${name}
+                PROPERTIES
+                INTERFACE_INCLUDE_DIRECTORIES "${CACTUS_INCLUDE_DIRS}"
+                INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${CACTUS_INCLUDE_DIRS}")
+
+            # Dependencies aren't written into .so as they should be, so we need to
+            # link explicitely
+            foreach(dep ${cactus_${name}_depends})
+                set_property(
+                    TARGET CACTUS::${name}
+                    APPEND PROPERTY INTERFACE_LINK_LIBRARIES
+                    CACTUS::${dep})
+            endforeach()
+        endif()
     endif()
-endfunction()
+endmacro()
 
 # Import all libs
-foreach(lib IN LISTS list_of_libs)
-    cactus_import_lib(${lib})
+foreach(lib ${cactus_requested_libs})
+    _cactus_import_lib(${lib})
 endforeach()
 
-# Print some debug info
-message(STATUS "Found the following CACTUS libraries:")
-foreach(lib ${list_of_libs})
-    message(STATUS "  ${lib}")
+# It looks like no version information is included in the headers
+set(CACTUS_VERSION "unknown")
+
+# Wrap things up
+set(CACTUS_LIBRARIES "")
+
+foreach(lib ${CACTUS_FIND_COMPONENTS})
+    find_package_handle_standard_args(
+        CACTUS_${lib}
+        FOUND_VAR CACTUS_${lib}_FOUND
+        REQUIRED_VARS ${cactus_${lib}_required_variables})
+
+    list(APPEND CACTUS_LIBRARIES ${CACTUS_${lib}_LIBRARY})
 endforeach()
+
+list(REMOVE_DUPLICATES CACTUS_LIBRARIES)
+
+find_package_handle_standard_args(
+    CACTUS
+    FOUND_VAR CACTUS_FOUND
+    REQUIRED_VARS CACTUS_INCLUDE_DIRS CACTUS_LIBRARIES
+    VERSION_VAR CACTUS_VERSION
+    HANDLE_COMPONENTS)
+
+# i2o requires an additional definition
```

Comment:
Does CACTUS need and/or provide `i2o` ? Aren't these line a leftover from `xDAQ` ?